src/ProjectNeon/Assets/Scripts/HPBarController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class HPBarController : MonoBehaviour
+{
+    #region Parametrs
+    [SerializeField] Image barImage;
+    [SerializeField] Text barTextValue;
+
+    int maxHP = 100;
+
+    public int currentHP { get; private set; }
+    public int MaxHP
+    {
+        set
+        {
+            maxHP = value;
+            UpdateUI();
+        }
+        get
+        {
+            return maxHP;
+        }
+    }
+
+    public delegate void Death();
+    public event Death OnDeath;
+    #endregion
+    #region Unity methods
+    private void Start()
+    {
+        currentHP = maxHP;
+        UpdateUI();
+    }
+    #endregion
+    #region Public methods
+    public void UpdateMaxHP(int _maxHP)
```

Comment:
Since this adds/substracts, we need a name that is consistent. Maybe this name can be `ChangeMaxHP` to match the following method?